CMakeLists.txt

```diff
@@ -14,10 +14,11 @@ project(${SOLUTION_NAME} LANGUAGES CXX VERSION ${SOLUTION_NAME_VERSION})
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_BUILD_TYPE Debug)
-set(CMAKE_CXX_FLAGS_DEBUG "-O1 -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_DEBUG "-g -O1 -Wall -Wextra")
```

Comment:
this is needed for gdb

Reply:
I think the cmake's gcc toolchain configuration already includes `-g` in its flags. Have you tried building with `-DCMAKE_BUILD_TYPE=Debug`? That should give you a build with the correct flags already, without the need of adding them yourself.

Reply:
I remember explicitly defining this variable to add some optimization and extra errors.

Some of the tests are heavily relying on return value optimization and in one case is the difference between taking 4 seconds vs 20 seconds. I'll look at whether there is a more modern way of adding the flags.

Reply:
Using `-DCMAKE_BUILD_TYPE=Release` should take you to `-O3` land. My point is only to avoid adding specific flags in the `CMakeLists.txt` file and use more of the flexibility given by cmake's options to produce builds at different optimization/quality levels.